msgpack-core/src/main/java/io/zeebe/msgpack/spec/MsgPackToken.java

```diff
@@ -64,15 +62,15 @@ public DirectBuffer getValueBuffer() {
   }
 
   public void setValue(DirectBuffer buffer, int offset, int length) {
-    if (offset < buffer.capacity() && offset + length <= buffer.capacity()) {
+    if (length == 0) {
+      valueBuffer.wrap(0, 0);
```

Comment:
This could cause problems if buffer has 0 capacity leading to original exception.

And while it is impossible in current callers of setValue() to be the case, I would suggest to avoid the need to remember such detail in setValue preconditions altogether, in case it would be invoked from somewhere else.

Reply:
Nope, i was wrong. Seems wrapping buffer with capacity 0 at index 0 takes different path than it is at wrapping buffer with capacity 1 at index 1.